src/formatters/ocaml.ts

```diff
@@ -0,0 +1,37 @@
+import { execSync } from "child_process";
+import * as fs from "fs";
+import * as path from "path";
+import { v4 as uuidv4 } from "uuid";
+import * as vscode from "vscode";
+import { getFullTextRange } from "./utils";
+
+export function register() {
+  const configuration = vscode.workspace.getConfiguration("reason");
+  const rootPath = vscode.workspace.rootPath || "";
+
+  vscode.languages.registerDocumentFormattingEditProvider(
+    { scheme: "file", language: "ocaml" },
+    {
+      provideDocumentFormattingEdits(_document: vscode.TextDocument): vscode.TextEdit[] {
+        const formatterPath = configuration.get<string | undefined>("path.ocamlformat");
+        const formatter = formatterPath ? path.resolve(rootPath, formatterPath) : "ocamlformat";
+        const textEditor = vscode.window.activeTextEditor;
+
+        if (textEditor) {
+          const tempFileName = `/tmp/vscode-reasonml-${uuidv4()}.ml`;
```

Comment:
This should be using `os.tmpdir()` instead of `/tmp` - otherwise it won't work correctly on Windows.

Reply:
I didn't think about windows at all. Will try to setup vm and test it there this week

Reply:
Thanks @rusty-key !